tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,463 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/pod"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/volume"
+	sc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/storage/v1/storageclass"
+	"github.com/openebs/dynamic-localpv-provisioner/tests/disk"
+)
+
+const (
+	// DiskImageSize is the default file size(1GB) used while creating backing image
+	DiskImageSize = 1073741824
+)
+
+var _ = Describe("TEST HOSTPATH XFS QUOTA LOCAL PV WITH NON-XFS FILESYSTEM", func() {
+	var (
+		pvcObj        *corev1.PersistentVolumeClaim
+		scObj         *storagev1.StorageClass
+		accessModes   = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+		capacity      = "5M"
+		podName       = "busybox-hostpath"
+		label         = "demo=hostpath-pod"
+		createdPvc    *corev1.PersistentVolumeClaim
+		pvcName       = "pvc-hp"
+		scNamePrefix  = "sc-hp-xfs"
+		scName        string
+		podObj        *corev1.Pod
+		labelselector = map[string]string{
+			"demo": "hostpath-pod",
+		}
+	)
+
+	When("StorageClass with valid xfs quota parameters is created", func() {
+		It("should create a StorageClass", func() {
+			By("building a StorageClass")
+			scObj, err = sc.NewStorageClass(
+				sc.WithGenerateName(scNamePrefix),
+				sc.WithLabels(map[string]string{
+					"openebs.io/test-sc": "true",
+				}),
+				sc.WithLocalPV(),
+				sc.WithHostpath("/var/openebs/integration-test"),
```

Comment:
`/var/openebs/integration-test` is being used at other places also. lets use a constant for this value